addons/l10n_ch/models/mail_template.py

```diff
@@ -16,28 +16,26 @@ def generate_email(self, res_ids, fields):
         present on the invoice. Otherwise, no ISR attachment will be created, and
         the mail will only contain the invoice (as defined in the mother method).
         """
-        rslt = super(MailTemplate, self).generate_email(res_ids, fields)
+        result = super(MailTemplate, self).generate_email(res_ids, fields)
+        if self.model != 'account.move':
+            return result
 
         multi_mode = True
         if isinstance(res_ids, int):
             res_ids = [res_ids]
             multi_mode = False
 
-        res_ids_to_templates = self.get_email_template(res_ids)
-        for res_id in res_ids:
-            related_model = self.env[self.model_id.model].browse(res_id)
+        for record in self.env[self.model].browse(res_ids):
+            if record.l10n_ch_isr_valid:
+                # We add an attachment containing the ISR
+                report_name = 'ISR-' + self._render_field('report_name', record.ids, compute_lang=True)[record.id] + '.pdf'
 
-            if related_model._name == 'account.move' and related_model.l10n_ch_isr_valid:
-                #We add an attachment containing the ISR
-                template = res_ids_to_templates[res_id]
-                report_name = 'ISR-' + self._render_template(template.report_name, template.model, res_id) + '.pdf'
-
-                pdf = self.env.ref('l10n_ch.l10n_ch_isr_report').render_qweb_pdf([res_id])[0]
+                pdf = self.env.ref('l10n_ch.l10n_ch_isr_report').render_qweb_pdf(record.ids)[0]
                 pdf = base64.b64encode(pdf)
 
-                attachments_list = multi_mode and rslt[res_id].get('attachments', False) or rslt.get('attachments', False)
+                attachments_list = multi_mode and result[record.id].get('attachments', False) or result.get('attachments', False)
                 if attachments_list:
                     attachments_list.append((report_name, pdf))
                 else:
-                    rslt[res_id]['attachments'] = [(report_name, pdf)]
-        return rslt
+                    result[record.id]['attachments'] = [(report_name, pdf)]
+        return result
```

Comment:
Well result is already correctly multi-mode'ed normally as it comes from calling super. There is a wrong line however (result[record.id]['attachments']) that requires a fix because it is not multi mode friendly.

That's why I will remove the multi mode soon :D . Overriding it is a PITA.